src/proxy/outbound.rs

```diff
@@ -260,9 +264,18 @@ impl OutboundConnection {
                 // Create a TCP connection to upstream
                 let mut outbound = TcpStream::connect(req.gateway).await?;
                 // Proxying data between downstrean and upstream
-                relay(&mut stream, &mut outbound).await?;
-
-                Ok(())
+                match relay(&mut stream, &mut outbound, self.cfg.zero_copy_enabled).await {
+                    // Connection closed with count of bytes transferred between streams
+                    Ok(Some(r)) => {
```

Comment:
```suggestion
                    Ok(Some((sent, recv))) => {
```

instead of r.0 and r.1 is more clear